src/redux/sagas/motions/expenditures/releaseExpenditureStageMotion.ts

```diff
@@ -0,0 +1,144 @@
+  const { createMotion /* annotationMotionMessage */ } = yield call(
+    createTransactionChannels,
+    id,
+    ['createMotion', 'annotationMotionMessage'],
+  );
+  const colonyManager = yield getColonyManager();
+  const colonyClient = yield colonyManager.getClient(
+    ClientType.ColonyClient,
+    colonyAddress,
+  );
+
+  try {
+    const stagedExpenditureClient = yield call(
+      [colonyManager, colonyManager.getClient],
+      ClientType.StagedExpenditureClient,
+      colonyAddress,
+    );
+
+    const [permissionDomainId, childSkillIndex] = yield getPermissionProofs(
+      colonyClient,
+      expenditure.nativeDomainId,
+      ColonyRole.Arbitration,
+      stagedExpenditureAddress,
+    );
+
+    const { skillId } = yield call(
+      [colonyClient, colonyClient.getDomain],
+      Id.RootDomain,
+    );
+
+    const { key, value, branchMask, siblings } = yield call(
+      [colonyClient, colonyClient.getReputation],
+      skillId,
+      ADDRESS_ZERO,
+    );
+
+    const encodedReleaseStagedPaymentAction =
+      yield stagedExpenditureClient.interface.encodeFunctionData(
+        'releaseStagedPayment(uint256,uint256,uint256,uint256,address[])',
+        [
+          permissionDomainId,
+          childSkillIndex,
+          expenditure.nativeId,
+          slotId,
+          tokenAddresses,
+        ],
+      );
+
+    yield createGroupTransaction(createMotion, batchId, meta, {
+      context: ClientType.VotingReputationClient,
+      methodName: 'createMotion',
+      identifier: colonyAddress,
+      params: [
+        motionDomainId,
+        childSkillIndex,
+        stagedExpenditureClient.address,
+        encodedReleaseStagedPaymentAction,
+        key,
+        value,
+        branchMask,
+        siblings,
+      ],
+      group: {
+        title: { id: 'transaction.group.createMotion.title' },
+        description: {
+          id: 'transaction.group.createMotion.description',
+        },
+      },
+    });
+
+    yield initiateTransaction({ id: createMotion.id });
+
+    const {
+      payload: { hash: txHash },
+    } = yield takeFrom(
+      createMotion.channel,
+      ActionTypes.TRANSACTION_HASH_RECEIVED,
+    );
+
+    setTxHash?.(txHash);
+
+    const { type } = yield call(waitForTxResult, createMotion.channel);
+
+    if (type === ActionTypes.TRANSACTION_SUCCEEDED) {
+      yield put<Action<ActionTypes.MOTION_RELEASE_EXPENDITURE_STAGE_SUCCESS>>({
+        type: ActionTypes.MOTION_RELEASE_EXPENDITURE_STAGE_SUCCESS,
+        meta,
+      });
+    }
+  } catch (error) {
+    console.error(error);
+    yield put<Action<ActionTypes.MOTION_RELEASE_EXPENDITURE_STAGE_ERROR>>({
+      type: ActionTypes.MOTION_RELEASE_EXPENDITURE_STAGE_ERROR,
+      payload: {
+        name: ActionTypes.MOTION_RELEASE_EXPENDITURE_STAGE_ERROR,
+        message: JSON.stringify(error),
+      },
+      meta,
+      error: true,
+    });
+  } finally {
+    createMotion.channel.close();
```

Comment:
Also down here we need to put the tx hash on the url params. This is a fairly recent product requirement but it enables users to see the txhash and come back to it in the new ui. I have done this for all other sagas, for reference.